src/include/btree.h

```diff
@@ -97,6 +97,12 @@ struct __wt_btree {
 	uint64_t maxmempage;		/* In-memory page max size */
 	uint64_t splitmempage;		/* In-memory split trigger size */
 
+#define	BTREE_DBG_COMMITTS_ALWAYS	0x0001
+#define	BTREE_DBG_COMMITTS_NEVER	0x0002
+#define	BTREE_DBG_READTS_ALWAYS		0x0004
+#define	BTREE_DBG_READTS_NEVER		0x0008
```

Comment:
I think these names could be better:
 * The should have a `WT_` prefix
 * Could you use `DEBUG` instead of `DBG`? I find that easier to read.
 * Having the `TS` without a separator made me think you'd misspelt `commits` at first

I know there is a risk of them becoming very long, so don't mind if you'd like to push back.

I'm also a fan of having the default represented with an explicit flag, which would mean adding `WT_BTREE_DEBUG_COMMIT_TS_NONE` and `WT_BTREE_DEBUG_READ_TS_NONE` (or whichever form you end up choosing for the common prefix). I don't think that adds to the overhead - we are adding a flag check to common paths regardless.

Reply:
I did not add in the `_NONE` settings because there is no place to use them and having and setting flags that are never used in code is confusing.  And adding code to check it when that isn't necessary also didn't appeal to me.  I took all the other suggestions.

There are a few other non-flag locations that don't prepend the structure name, such as for direct_io.  So to prevent unpleasantly long names, I went with `WT_DEBUG_COMMIT_TS_ALWAYS`, etc.